File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        final Set<Integer> s = new HashSet<>();
        final int size = nums.length;
        for (int i=0; i<size; i++) {
            if(s.contains(nums[i])) return true;
            s.add(nums[i]);
        }
        return false;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int size = nums.size();
        vector<int> res(size);

        int product = 1;
        for (int i=0; i<size; i++) {
            res[i] = product;
            product *= nums[i];
        }

        product = 1;
        for (int i=size-1; i>=0; i--) {
            res[i] *= product;
            product *= nums[i];
        }

        return res;
    }
};
